### PR TITLE
Altered demo notebooks for Kueue default & mTLS default

### PR DESCRIPTION
**Pull request by Bobbins228**
# Issue link

[RHOAIENG-1913](https://issues.redhat.com/browse/RHOAIENG-1913)
# What changes have been made

Updated existing notebooks to use Kueue by default and have notes for setup.
Removed mention of local_interactive
Removed InstaScale NB and mentions of MCAD from other NBs
# Verification steps

Run the demo notebook when #470 
## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: demo-notebooks/guided-demos/2_job_client.ipynb [hunk 1]
```diff
@@ -52,7 +52,13 @@
     "    min_memory=4,\n",
     "    max_memory=4,\n",
     "    num_gpus=0,\n",
+<<<<<<< HEAD:demo-notebooks/guided-demos/2_job_client.ipynb
     "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\"\n",
+=======
+    "    mcad=True,\n",
+    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    instascale=False\n",
+>>>>>>> 583747f (Added demo notebook for Kueue by default):demo-notebooks/guided-demos/preview_nbs/2_basic_jobs.ipynb
```
Comment (KPostOffice) [inline demo-notebooks/guided-demos/2_job_client.ipynb:61]: some left over diffs from merge conflict

File: demo-notebooks/guided-demos/2_job_client.ipynb [hunk 2]
```diff
@@ -252,6 +258,77 @@
...
+   "outputs": [],
+   "source": [
+    "job.cancel()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0d41b90e",
+>>>>>>> 583747f (Added demo notebook for Kueue by default):demo-notebooks/guided-demos/preview_nbs/2_basic_jobs.ipynb
```
Comment (KPostOffice) [inline demo-notebooks/guided-demos/2_job_client.ipynb:331]: more diff

Comment (ChristianZaccaria) [reply demo-notebooks/guided-demos/2_job_client.ipynb]: Good catch, that was causing to not be able to even open the file: 
image

File: demo-notebooks/additional-demos/hf_interactive.ipynb [hunk 3]
```diff
@@ -89,7 +91,8 @@
     }
    ],
    "source": [
-    "# Create our cluster and submit appwrapper\n",
+    "# Create our cluster and submit\n",
+    "# The SDK will try to find the name of your default local queue based on the annotation \"kueue.x-k8s.io/default-queue\": \"true\"\n",
```
Comment (Fiona-Waters) [inline demo-notebooks/additional-demos/hf_interactive.ipynb:95]: Does specifying it manually override the use of the annotation?
```suggestion
    "# The SDK will try to find the name of your default local queue based on the annotation \"kueue.x-k8s.io/default-queue\": \"true\"\n" unless you specify the local queue manually below,
```

File: demo-notebooks/guided-demos/0_basic_ray.ipynb [hunk 4]
```diff
@@ -58,18 +60,19 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Create and configure our cluster object (and appwrapper)\n",
+    "# Create and configure our cluster object\n",
+    "# The SDK will try to find the name of your default local queue based on the annotation \"kueue.x-k8s.io/default-queue\": \"true\"\n",
```
Comment (Fiona-Waters) [inline demo-notebooks/guided-demos/0_basic_ray.ipynb:64]: Same question

File: demo-notebooks/guided-demos/3_basic_interactive.ipynb [hunk 5]
```diff
@@ -5,7 +5,7 @@
    "id": "bbc21043",
    "metadata": {},
    "source": [
-    "In this fourth and final notebook, we will go over how to leverage the SDK to directly work interactively with a Ray cluster during development."
+    "In this fourth notebook, we will go over how to leverage the SDK to directly work interactively with a Ray Cluster during development."
```
Comment (Fiona-Waters) [inline demo-notebooks/guided-demos/3_basic_interactive.ipynb:8]:
```suggestion
    "In this notebook, we will go over how to leverage the SDK to directly work interactively with a Ray Cluster during development."
```

File: demo-notebooks/guided-demos/3_basic_interactive.ipynb [hunk 6]
```diff
@@ -55,20 +57,21 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Create and configure our cluster object (and appwrapper)\n",
+    "# Create and configure our cluster object\n",
+    "# The SDK will try to find the name of your default local queue based on the annotation \"kueue.x-k8s.io/default-queue\": \"true\"\n",
```
Comment (Fiona-Waters) [inline demo-notebooks/guided-demos/3_basic_interactive.ipynb:61]: same question about specifying local queue/using annotation - applies everywhere this is mentioned

File: demo-notebooks/guided-demos/notebook-ex-outputs/2_cluster_job_client.ipynb [hunk 7]
```diff
@@ -0,0 +1,223 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this third demo we will go over the basics of the Ray Job Submission Client in the SDK"
```
Comment (Fiona-Waters) [inline demo-notebooks/guided-demos/notebook-ex-outputs/2_cluster_job_client.ipynb:7]:
```suggestion
    "In this demo we will go over the basics of the Ray Job Submission Client in the SDK"
```

File: demo-notebooks/guided-demos/3_basic_interactive.ipynb [hunk 8]
```diff
@@ -125,6 +128,19 @@
...
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9436436",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from codeflare_sdk import generate_cert\n",
+    "# Export the environment variables to enable TLS\n",
```
Comment (Fiona-Waters) [inline demo-notebooks/guided-demos/3_basic_interactive.ipynb:139]: I think this should be updated to reflect these functions more accurately. This is in 3 different places in the PR.
```suggestion
    "# Create required TLS cert and export the environment variables to enable TLS\n",
```

File: demo-notebooks/additional-demos/local_interactive.ipynb [hunk 9]
```diff
@@ -49,12 +49,10 @@
    "outputs": [],
    "source": [
     "# Create our cluster and submit appwrapper\n",
-    "namespace = \"default\"\n",
+    "namespace = \"default\" # Update to your namespace\n",
```
Comment (Fiona-Waters) [inline demo-notebooks/additional-demos/local_interactive.ipynb:52]: Should we be consistent in how we provide the namespace in each notebook? Sometimes we define it before cluster config and other times we don't.

**Review by Fiona-Waters — COMMENTED** [review]
Currently setting up to run some of the notebooks on openshift. Just some comments first. Thanks @Bobbins228

File: demo-notebooks/additional-demos/local_interactive.ipynb (hunk 9 above)
Comment (Bobbins228) [reply demo-notebooks/additional-demos/local_interactive.ipynb:52]: For demos where we define it before config we use the `namespace` & `cluster_name` vars for generating certs later.

File: demo-notebooks/guided-demos/0_basic_ray.ipynb [hunk 10]
```diff
@@ -45,10 +45,12 @@
...
    "metadata": {},
    "source": [
-    "Here, we want to define our cluster by specifying the resources we require for our batch workload. Below, we define our cluster object (which generates a corresponding AppWrapper).\n",
+    "Here, we want to define our cluster by specifying the resources we require for our batch workload. Below, we define our cluster object (which generates a corresponding RayCluster).\n",
     "\n",
     "NOTE: We must specify the `image` which will be used in our RayCluster, we recommend you bring your own image which suits your purposes. \n",
-    "The example here is a community image."
+    "The example here is a community image.\n",
+    "\n",
+    "NOTE: By default the SDK uses Kueue as it's scheduling solution to use MCAD set the `mcad=True` option in `ClusterConfiguration`"
```
Comment (ChughShilpa) [inline demo-notebooks/guided-demos/0_basic_ray.ipynb:53]: I think it is better to mention that MCAD component is in Tech Preview along wherever we have mentioned 'mcad=True'

Comment (ChughShilpa) [reply demo-notebooks/guided-demos/0_basic_ray.ipynb]: I can see the change only in 'guided-demos' notebook. This should be updated in each notebook wherever enabling of mcad is mentioned

**Comment by ChughShilpa** [comment]
Just wondering why we are not adding ` write_to_file` parameter in the cluster configuration in any of the demo notebooks. 
@Bobbins228

File: demo-notebooks/guided-demos/0_basic_ray.ipynb (hunk 10 above)
Comment (Bobbins228) [reply demo-notebooks/guided-demos/0_basic_ray.ipynb]: Oops I missed one thanks

**Comment by Bobbins228** [comment]
> Just wondering why we are not adding ` write_to_file` parameter in the cluster configuration in any of the demo notebooks. @Bobbins228

This is because by default we do not write to a file but I could include it with a comment if that sounds like it would be beneficial to new users

File: demo-notebooks/additional-demos/hf_interactive.ipynb [hunk 11]
```diff
@@ -89,7 +91,8 @@
     }
    ],
    "source": [
-    "# Create our cluster and submit appwrapper\n",
+    "# Create our cluster and submit\n",
+    "# The SDK will try to find the name of your default local queue based on the annotation \"kueue.x-k8s.io/default-queue\": \"true\" unless you specify the local queue manually below\n",
     "cluster = Cluster(ClusterConfiguration(name='hfgputest', \n",
     "                                       namespace=\"default\",\n",
```
Comment (ChughShilpa) [inline demo-notebooks/additional-demos/hf_interactive.ipynb:97]: A small nitpick 
 # Update to your namespace is missing here

File: demo-notebooks/additional-demos/local_interactive.ipynb [hunk 12]
```diff
@@ -48,13 +48,12 @@
...
-    "# Create our cluster and submit appwrapper\n",
-    "namespace = \"default\"\n",
+    "# Create and submit our cluster\n",
+    "# The SDK will try to find the name of your default local queue based on the annotation \"kueue.x-k8s.io/default-queue\": \"true\" unless you specify the local queue manually below\n",
+    "namespace = \"default\" # Update to your namespace\n",
     "cluster_name = \"hfgputest-1\"\n",
-    "local_interactive = True\n",
     "\n",
-    "cluster = Cluster(ClusterConfiguration(local_interactive=local_interactive,\n",
-    "                                       namespace=namespace,\n",
```
Comment (ChughShilpa) [inline demo-notebooks/additional-demos/local_interactive.ipynb:57]: same here

File: demo-notebooks/guided-demos/0_basic_ray.ipynb (hunk 10 above)
Comment (dgrove-oss) [reply demo-notebooks/guided-demos/0_basic_ray.ipynb]: I would recommend removing user visible references to MCAD entirely from the demo notebooks.  We'll want to add in an example of using AppWrappers but that's not an alternative to Kueue in the same way MCAD was.

Comment (Bobbins228) [reply demo-notebooks/guided-demos/0_basic_ray.ipynb]: @dgrove-oss Sure thing

Comment (Bobbins228) [reply demo-notebooks/guided-demos/0_basic_ray.ipynb]: What about the InstaScale demo?
MCAD v1 and InstaScale are still tech preview

Comment (dgrove-oss) [reply demo-notebooks/guided-demos/0_basic_ray.ipynb]: My understanding is that we are pulling InstaScale as well.  I'd also recommend cutting InstaScale from the demo notebooks.

File: demo-notebooks/additional-demos/local_interactive.ipynb [hunk 13]
```diff
@@ -63,6 +62,7 @@
     "                                       max_memory=4,\n",
     "                                       num_gpus=0,\n",
     "                                       image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "                                       write_to_file=False, # When enabled Ray Cluster yaml files are written to /HOME/.codeflare/resources \n",
     "                                       instascale=False,\n",
```
Comment (ChughShilpa) [inline demo-notebooks/additional-demos/local_interactive.ipynb:66]: Instascale should be removed from here as well

File: demo-notebooks/additional-demos/ray_job_client.ipynb [hunk 14]
```diff
@@ -43,16 +42,18 @@
...
     "    num_workers=2,\n",
     "    min_cpus=1,\n",
     "    max_cpus=1,\n",
     "    min_memory=4,\n",
     "    max_memory=4,\n",
     "    num_gpus=0,\n",
-    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\"\n",
+    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    write_to_file=False # When enabled Ray Cluster yaml files are written to /HOME/.codeflare/resources \n",
     "))"
```
Comment (ChughShilpa) [inline demo-notebooks/additional-demos/ray_job_client.ipynb:57]: local_queue=\"local-queue-name\" parameter is missing

File: demo-notebooks/guided-demos/1_cluster_job_client.ipynb [hunk 15]
```diff
@@ -43,16 +41,18 @@
...
     "    num_workers=2,\n",
     "    min_cpus=1,\n",
     "    max_cpus=1,\n",
     "    min_memory=4,\n",
     "    max_memory=4,\n",
     "    num_gpus=0,\n",
-    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\"\n",
+    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    write_to_file=False # When enabled Ray Cluster yaml files are written to /HOME/.codeflare/resources \n",
     "))"
```
Comment (ChughShilpa) [inline demo-notebooks/guided-demos/1_cluster_job_client.ipynb:56]: local_queue=\"local-queue-name\" parameter is missing

File: demo-notebooks/guided-demos/notebook-ex-outputs/1_cluster_job_client.ipynb [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+    "    namespace='default', # Update to your namespace\n",
+    "    num_workers=2,\n",
+    "    min_cpus=1,\n",
+    "    max_cpus=1,\n",
+    "    min_memory=4,\n",
+    "    max_memory=4,\n",
+    "    num_gpus=0,\n",
+    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    write_to_file=False # When enabled Ray Cluster yaml files are written to /HOME/.codeflare/resources \n",
+    "))"
```
Comment (ChughShilpa) [inline demo-notebooks/guided-demos/notebook-ex-outputs/1_cluster_job_client.ipynb:56]: local_queue=\"local-queue-name\" parameter is missing

File: demo-notebooks/guided-demos/preview_nbs/1_cluster_job_client.ipynb [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+    "    namespace='default', # Update to your namespace\n",
+    "    num_workers=2,\n",
+    "    min_cpus=1,\n",
+    "    max_cpus=1,\n",
+    "    min_memory=4,\n",
+    "    max_memory=4,\n",
+    "    num_gpus=0,\n",
+    "    image=\"quay.io/project-codeflare/ray:latest-py39-cu118\",\n",
+    "    write_to_file=False # When enabled Ray Cluster yaml files are written to /HOME/.codeflare/resources \n",
+    "))"
```
Comment (ChughShilpa) [inline demo-notebooks/guided-demos/preview_nbs/1_cluster_job_client.ipynb:56]: local_queue=\"local-queue-name\" parameter is missing

**Review by Fiona-Waters — COMMENTED** [review]
I've tested some of the notebooks on openshift and they work as expected. Just one small comment but otherwise looks great to me. Great work Mark!

**Review by ChughShilpa — APPROVED** [review]
Tested some notebooks and it worked as expected

**Pull request merged**